test/e2e-kind.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Tekton Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a cop y of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script calls out to scripts in tektoncd/plumbing to setup a cluster
+# and deploy Tekton Pipelines to it for running integration tests.
+
+source $(git rev-parse --show-toplevel)/test/e2e-common.sh
+
+# Script entry point.
+
+initialize $@ --cloud-provider kind
+
+header "Setting up environment"
+
+install_pipeline_crd
+
+failed=0
+
+# Run the integration tests
+header "Running Go e2e tests"
+go_test_e2e -timeout=40m ./test/... -run="TestTaskRunPipelineRunCancel|TestDAGPipelineRun" || failed=1
+
+# Run these _after_ the integration tests b/c they don't quite work all the way
```

Comment:
Should we add a `(( failed )) && fail_test` here too, so we fail fast if the first batch of tests fail, and don't bother to run the rest?

Reply:
Sure. Should we add `(( failed )) && fail_test` to e2e-tests.sh too? I think it also needs this.

Reply:
Oh I didn't realize this was copied from e2e-tests.sh. In that case keeping the logic the same seems good, and we can open another PR to update both of them to fail faster, and discuss whether we want that.